pkg/agent/route/route_linux.go

```diff
@@ -1344,31 +1345,31 @@ func (c *Client) AddClusterIPRoute(svcIP net.IP) error {
 		return nil
 	}
 
+	// If the route does not exist or exists but its destination CIDR cannot contain the ClusterIP, get the calculated
+	// CIDR from serviceCIDRController.
 	var newClusterIPCIDR *net.IPNet
-	var err error
-	if curClusterIPCIDR != nil {
-		// If the route exists and its destination CIDR doesn't contain the ClusterIP, generate a new destination CIDR by
-		// enlarging the current destination CIDR with the ClusterIP.
-		if newClusterIPCIDR, err = util.ExtendCIDRWithIP(curClusterIPCIDR, svcIP); err != nil {
-			return fmt.Errorf("enlarge the destination CIDR with an error: %w", err)
+	if err := wait.PollImmediate(time.Millisecond*100, 2*time.Second, func() (bool, error) {
+		cidr := c.serviceCIDRController.GetClusterIPCIDR(isIPv6)
+		if cidr.Contains(svcIP) {
+			newClusterIPCIDR = cidr
+			return true, nil
 		}
-	} else {
-		// If the route doesn't exist, generate a new destination CIDR with the ClusterIP. Note that, this is the first
-		// ClusterIP since the route doesn't exist.
-		newClusterIPCIDR = &net.IPNet{IP: svcIP, Mask: net.CIDRMask(mask, mask)}
+		return false, nil
+	}); err != nil {
+		return fmt.Errorf("failed to get the latest calculated ClusterIP CIDR which contains the ClusterIP: %w", err)
```

Comment:
Instead of polling here, the route client could register a handler to serviceCIDR provider and run a goroutine in `initServiceIPRoutes` which keeps reconciling the routes for serviceCIDR.

Reply:
I didn't get your idea totally. In the latest commit, my implements are:
- Register handler function in `initServiceIPRoutes` to ServiceProvider
- When ServiceProvider updates Service CIDR, call every handler function
- In handler function for AntreaProxy, install ServiceCIDR route.